### DISCUSSION TITLE
How to authenticate password of a current logged in user ?

### DISCUSSION THREAD

**Discussion by geminigeek** [payloadcms/payload#3167] [Q&A]
hi,

i want user to confirm the password before letting a critical setting to turn on/off, i was using an exposed function like this (below) , now this function is removed or refactored , i am trying to update my old version of payload and it seems to be a breaking change , can anyone help me how achieve this in the current payload version ?

```
const authResult = await userDoc.authenticate("userProvidedPassword")
```

**Comment by sebastianspiller** [comment]
Hi there,

we would like to allow the user to change her password without the need to logout and click on reset password. Currently, there is no such operation. Can we / I build that and make a PR or shouldn't it be allowed by design?

Thx for joining the discussion. :)
